File: ios/StillMomentTests/Mocks/MockSoundscapeResolver.swift
//
//  MockSoundscapeResolver.swift
//  Still Moment
//
//  Test mock for SoundscapeResolverProtocol
//

import Foundation
@testable import StillMoment

final class MockSoundscapeResolver: SoundscapeResolverProtocol {
    // MARK: - Configurable returns

    var stubbedResolveResults: [String: ResolvedSoundscape] = [:]
    var stubbedAudioURLs: [String: URL] = [:]
    var stubbedAllAvailable: [ResolvedSoundscape] = []
    var shouldThrowOnResolveURL: Error?

    // MARK: - Recorded calls

    var resolvedIds: [String] = []
    var resolvedAudioURLIds: [String] = []

    // MARK: - Protocol

    func resolve(id: String) -> ResolvedSoundscape? {
        self.resolvedIds.append(id)
        return self.stubbedResolveResults[id]
    }

    func resolveAudioURL(id: String) throws -> URL {
        self.resolvedAudioURLIds.append(id)
        if let error = self.shouldThrowOnResolveURL { throw error }
        guard let url = self.stubbedAudioURLs[id] else {
            throw AudioServiceError.soundFileNotFound
        }
        return url
    }

    func allAvailable() -> [ResolvedSoundscape] {
        self.stubbedAllAvailable
    }
}

File: android/app/src/test/kotlin/com/stillmoment/testutil/MockSoundscapeResolver.kt
package com.stillmoment.testutil

import com.stillmoment.domain.models.ResolvedSoundscape
import com.stillmoment.domain.services.SoundscapeResolverProtocol

/**
 * Test mock for SoundscapeResolverProtocol.
 *
 * Returns pre-configured results for resolve() and allAvailable().
 */
class MockSoundscapeResolver(
    private val resolveResult: Map<String, ResolvedSoundscape> = emptyMap(),
    private val allAvailableResult: List<ResolvedSoundscape> = emptyList()
) : SoundscapeResolverProtocol {
    override fun resolve(id: String): ResolvedSoundscape? = resolveResult[id]
    override fun allAvailable(): List<ResolvedSoundscape> = allAvailableResult
}
